src/sinks/datadog/metrics/integration_tests.rs

```diff
@@ -104,16 +126,105 @@ async fn smoke() {
 
     assert!(output.len() == 1, "Should have received a response");
 
-    let val = output.first().unwrap();
+    let request = output.first().unwrap();
+
+    match request.0.uri.path() {
+        SERIES_V1_PATH => validate_json(request),
+        SERIES_V2_PATH => validate_protobuf(request),
+        _ => panic!("Unexpected request type received!"),
+    }
+}
 
+fn validate_common(request: &(Parts, Bytes)) {
+    assert_eq!(request.0.headers.get("DD-API-KEY").unwrap(), "atoken");
+    assert!(request.0.headers.contains_key("DD-Agent-Payload"));
+}
+
+fn validate_protobuf(request: &(Parts, Bytes)) {
     assert_eq!(
-        val.0.headers.get("Content-Type").unwrap(),
+        request.0.headers.get("Content-Type").unwrap(),
+        "application/x-protobuf"
+    );
+
+    validate_common(request);
+
+    let compressed_payload = request.1.to_vec();
+    let payload = decompress_payload(compressed_payload).expect("Could not decompress payload");
+    let frame = Bytes::copy_from_slice(&payload);
+
+    let payload =
+        ddmetric_proto::MetricPayload::decode(frame).expect("Could not decode protobuf frame");
+
+    let series = payload.series;
+
+    assert!(!series.is_empty());
+
+    // check metrics are sorted by name, which helps HTTP compression
+    let metric_names: Vec<String> = series.iter().map(|serie| serie.metric.clone()).collect();
+    let mut sorted_names = metric_names.clone();
+    sorted_names.sort();
+    assert_eq!(metric_names, sorted_names);
+
+    series.iter().for_each(|serie| {
+        // name
+        assert!(serie.metric.starts_with("foo.counter_"));
+
+        // type
+        assert_eq!(
+            serie.r#type(),
+            ddmetric_proto::metric_payload::MetricType::Count
+        );
+
+        // resources
+        serie
+            .resources
+            .iter()
+            .for_each(|resource| match resource.r#type.as_str() {
+                "host" => assert_eq!(resource.name.as_str(), "a_host"),
+                "device" => assert_eq!(resource.name.as_str(), "a_device"),
+                _ => panic!("Unexpected resource found!"),
+            });
+
+        // source_type_name
+        assert_eq!(serie.source_type_name, "a_name");
+
+        // tags
+        assert_eq!(serie.tags.len(), 1);
+        assert_eq!(serie.tags.first().unwrap(), "cool_tag_name:i_know_right");
+
+        // unit
+        assert!(serie.unit.is_empty());
+
+        // interval
+        assert_eq!(serie.interval, 0);
```

Comment:
💭 It would be good to exercise more code paths in our tests. Specifically, testing (1) a counter with an interval specified, (2) a gauge, and (3) a set.